ntbs-integration-tests/NotificationPages/TestResultsEditPageTests.cs

```diff
@@ -0,0 +1,88 @@
+using Xunit;
+
+namespace ntbs_integration_tests.NotificationPages
+{
+    public class TestResultsEditPageTest : TestRunnerNotificationBase
+    {
+        protected override string NotificationSubPath => NotificationSubPaths.EditTestResults;
+
+        public TestResultsEditPageTest(NtbsWebApplicationFactory<Startup> factory) : base(factory)
+        {
+        }
+
+        [Fact]
+        public async Task IfMatchingLabResultsExist_DisplaysLabResults()
+        {
+            // Arrange
+            const int notificationId = Utilities.NOTIFIED_ID;
+
+            // Act
+            var response = await Client.GetAsync(GetCurrentPathForId(notificationId));
+
+            // Assert
+            var document = await GetDocumentAsync(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.NotNull(document.QuerySelector("#lab-results-summary"));
+            Assert.NotNull(document.QuerySelector("#specimens-details"));
+        }
+
+        [Fact]
+        public async Task IfMatchingLabResultsDoNotExist_DoNotDisplaysLabResults()
+        {
+            // Arrange
+            const int notificationId = Utilities.DRAFT_ID;
+
+            // Act
+            var response = await Client.GetAsync(GetCurrentPathForId(notificationId));
+
+            // Assert
+            var document = await GetDocumentAsync(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Null(document.QuerySelector("#lab-results-summary"));
+            Assert.Null(document.QuerySelector("#specimens-details"));
+        }
+
+        [Fact]
+        public async Task RedirectsToOverviewWithCorrectAnchorFragment_ForNotified()
+        {
+            // Arrange
+            const int id = Utilities.NOTIFIED_ID;
+            var url = GetCurrentPathForId(id);
+            var document = await GetDocumentForUrlAsync(url);
+
+            var formData = new Dictionary<string, string>
+            {
+                ["NotificationId"] = id.ToString(),
+                ["TestData.HasTestCarriedOut"] = "false"
+            };
+
+            // Act
+            var result = await Client.SendPostFormWithData(document, formData, url);
+
+            // Assert
+            var sectionAnchorId = OverviewSubPathToAnchorMap.GetOverviewAnchorId(NotificationSubPath);
+            result.AssertRedirectTo($"/Notifications/{id}#{sectionAnchorId}");
+        }
+
+        [Fact]
+        public async Task NotifiedPageHasReturnLinkToOverview()
+        {
+            // Arrange
+            const int id = Utilities.NOTIFIED_ID;
+            var url = GetCurrentPathForId(id);
+
+            // Act
+            var document = await GetDocumentForUrlAsync(url);
+
+            // Assert
+            var overviewLink = RouteHelper.GetNotificationOverviewPathWithSectionAnchor(id, NotificationSubPath);
+            Assert.NotNull(document.QuerySelector($"a[href='{overviewLink}']"));
+        }
+    }
+}
```

Comment:
I renamed this test results test class to test result edit tests, as that's the page they were testing - so I didn't actually add all of this